src/lib/recurringPurposes.js

```diff
@@ -1,18 +1,248 @@
-// @ts-check
 import { set } from 'lodash'
-import { OTHER_PURPOSE } from 'src/constants'
-import { setAutomaticPurpose } from 'src/lib/timeseries'
+import {
+  COMMUTE_PURPOSE,
+  HOME_ADDRESS_CATEGORY,
+  OTHER_PURPOSE,
+  WORK_ADDRESS_CATEGORY
+} from 'src/constants'
+import { CONTACTS_DOCTYPE } from 'src/doctypes'
+import {
+  getEndPlaceCoordinates,
+  getStartPlaceCoordinates,
+  setAutomaticPurpose
+} from 'src/lib/timeseries'
 import { getManualPurpose, getAutomaticPurpose } from 'src/lib/trips'
 import {
+  buildContactsWithGeoCoordinates,
   buildNewestRecurringTimeseriesQuery,
+  buildRecurringTimeseriesByStartAndEndPointRange,
   buildSettingsQuery,
   buildTimeseriesQueryByAccountIdAndDate,
-  builTimeserieQueryByAccountIdAndStartPlaceAndEndPlaceAndStartDateAndDistance,
+  queryContactByDocId,
   queryTimeserieByDocId
 } from 'src/queries/queries'
 
+import { models } from 'cozy-client'
 import log from 'cozy-logger'
 
+const { deltaLatitude, deltaLongitude, geodesicDistance } = models.geo
+
+const MAX_SPATIAL_THRESHOLD_M = 200
+
+const findContactsWithGeo = async client => {
+  const queryDef = buildContactsWithGeoCoordinates().definition
+  const contacts = await client.queryAll(queryDef)
+  return contacts
+}
+
+export const findClosestStartAndEnd = (timeserie, contacts) => {
+  const startCoordinates = getStartPlaceCoordinates(timeserie)
+  const endCoordinates = getEndPlaceCoordinates(timeserie)
+  let closestStart = {},
+    closestEnd = {}
+  let minStartDistance = Number.MAX_SAFE_INTEGER,
+    minEndDistance = Number.MAX_SAFE_INTEGER
+  for (const contact of contacts) {
+    for (const address of contact.address) {
+      const addressCoordinates = {
+        lon: address.geo.geo[0],
+        lat: address.geo.geo[1]
+      }
+      const startPlaceDistance = geodesicDistance(
+        startCoordinates,
+        addressCoordinates
+      )
+
+      const endPlaceDistance = geodesicDistance(
+        endCoordinates,
+        addressCoordinates
+      )
+
+      if (startPlaceDistance < minStartDistance) {
+        closestStart = {
```

Comment:
Does that mean, that if I only have one contact with coordinates, even if the contact if pretty distant, it'll be assigned to this closestStuff?